MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs

```diff
@@ -71,300 +71,321 @@ public UserDeveloperAPIModel(UserDataViewModel userData)
-                            return;
-                        }
-                        else
-                        {
-                            await this.CloseConnection(listenerContext, HttpStatusCode.NotFound, "Could not find the command specified");
-                            return;
-                        }
-                    }
-                }
-                else if (httpMethod.Equals(PutHttpMethod) || httpMethod.Equals(PatchHttpMethod))
-                {
-                    if (urlSegments.Count() == 2 && Guid.TryParse(urlSegments[1], out Guid ID))
-                    {
-                        CommandBase commandData = SerializerHelper.DeserializeAbstractFromString<CommandBase>(data);
-                        CommandBase matchedCommand = allCommands.FirstOrDefault(c => c.ID.Equals(ID));
-                        if (matchedCommand != null)
-                        {
-                            matchedCommand.IsEnabled = commandData.IsEnabled;
-
-                            await this.CloseConnection(listenerContext, HttpStatusCode.OK, SerializerHelper.SerializeToString(matchedCommand));
-                            return;
-                        }
-                        else
-                        {
-                            await this.CloseConnection(listenerContext, HttpStatusCode.NotFound, "Invalid data/could not find matching command");
-                            return;
-                        }
-                    }
-                }
+            return selectedCommand;
+        }
+
+        [Route("{commandID:guid}")]
+        [HttpPut, HttpPatch]
+        public CommandBase Update(Guid commandID, [FromBody] CommandBase commandData)
+        {
+            CommandBase selectedCommand = GetAllCommands().SingleOrDefault(c => c.ID == commandID);
+            if (selectedCommand == null)
+            {
+                throw new HttpResponseException(HttpStatusCode.NotFound);
             }
-            else if (urlSegments[0].Equals("spotify") && urlSegments.Count() >= 2)
+
+            selectedCommand.IsEnabled = commandData.IsEnabled;
+            return selectedCommand;
+        }
+
+        private List<CommandBase> GetAllCommands()
+        {
+            List<CommandBase> allCommands = new List<CommandBase>();
+            allCommands.AddRange(ChannelSession.Settings.ChatCommands);
+            allCommands.AddRange(ChannelSession.Settings.InteractiveCommands);
+            allCommands.AddRange(ChannelSession.Settings.EventCommands);
+            allCommands.AddRange(ChannelSession.Settings.TimerCommands);
+            allCommands.AddRange(ChannelSession.Settings.ActionGroupCommands);
+            allCommands.AddRange(ChannelSession.Settings.GameCommands);
+            return allCommands;
+        }
+    }
+
+    [RoutePrefix("api/spotify")]
+    public class SpotifyController : ApiController
```

Comment:
Let's just get rid of this API for now, we can eventually replace it with Song Requests.

Reply:
Removed.